PhysicsTools/PatAlgos/plugins/BaseMVAValueMapProducer.h

```diff
@@ -63,19 +73,47 @@ class BaseMVAValueMapProducer : public edm::stream::EDProducer<> {
       values_.resize(variablesOrder_.size());
       size_t i=0;
       for(const auto & v : variablesOrder_){
-	positions_[v]=i;
-	reader_.AddVariable(v,(&values_.front())+i);
-	i++;
+        positions_[v]=i;
+        if(tmva_) reader_->AddVariable(v,(&values_.front())+i);
+	    i++;
       }
 //      reader_.BookMVA(name_,iConfig.getParameter<edm::FileInPath>("weightFile").fullPath() );
-      reco::details::loadTMVAWeights(&reader_, name_, iConfig.getParameter<edm::FileInPath>("weightFile").fullPath());
-      produces<edm::ValueMap<float>>();
+    if(tmva_)
+    {
+        reco::details::loadTMVAWeights(reader_, name_, weightfilename_);
+    }else if(tf_)    {
+        tensorflow::setLogging("3");
+        graph_=tensorflow::loadGraphDef(weightfilename_);
```

Comment:
what is the typical size of this?
More than a few MB should be placed in a GlobalCache.
Based on about 10MB size of breg_training_2017.pb, I suspect that the size can be pretty large.

Reply:
for nanoaod this should not be a problem, but if someone want to make this module more performant it may make sense to do several things, including evaluating in batches.
An issue can be opened on this but I would not change it for this PR

Reply:
I think that updates can be made in a day or so, with enough effort; no need to delay for some future.

Reply:
I'm not sure if you are volunteering for it or what, in the next day or so I have exactly zero time for that.

Reply:
OK, this probably implies 9_4_AN_X and for 94X and some time to wait until this is implemented in the master

Reply:
I have no problem with changing the 94X PR to AN, as long as  this is ok with @fabiocos